presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraPartitionManager.java

```diff
@@ -126,15 +127,20 @@ public CassandraPartitionResult getPartitions(ConnectorTableHandle tableHandle,
         }
 
         Set<List<Object>> partitionKeysSet = getPartitionKeysSet(table, tupleDomain);
+        Set<List<Object>> clusteringKeysSet = getClusteringKeysSet(table, tupleDomain);
 
         // empty filter means, all partitions
         if (partitionKeysSet.isEmpty()) {
             return schemaProvider.getAllPartitions(table);
         }
 
         ImmutableList.Builder<ListenableFuture<List<CassandraPartition>>> getPartitionResults = ImmutableList.builder();
+
+        // todo: is the following nested for loop correct?
```

Comment:
As I noted above, I think you should store the clustering key information somewhere other than CassandraPartition; you aren't getting anything out of storing the clustering key info there b/c the underlying Cassandra partitions will be the same. If you're forced to because of how this code is structured, maybe the partitioning key part needs to be refactored...